x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/update/cleanup_stuck_in_progress_gaps.ts

```diff
@@ -0,0 +1,142 @@
+const MAX_RULES_TO_PROCESS = 100;
+const STUCK_GAP_THRESHOLD_HOURS = 12;
+
+interface CleanupStuckInProgressGapsParams {
+  rulesClientContext: RulesClientContext;
+  eventLogClient: IEventLogClient;
+  eventLogger: IEventLogger;
+  logger: Logger;
+  startDate: Date;
+}
+
+/**
+ * Cleanup stuck in-progress gaps that don't have corresponding backfills.
+ * Finds gaps with in_progress_intervals that were updated more than 12 hours ago,
+ * checks if backfills exist for them, and resets the in_progress_intervals if no backfills exist.
+ */
+export const cleanupStuckInProgressGaps = async ({
+  rulesClientContext,
+  eventLogClient,
+  eventLogger,
+  logger,
+  startDate,
+}: CleanupStuckInProgressGapsParams): Promise<void> => {
+  try {
+    const now = new Date();
+    // Calculate cutoff time: now - 12 hours
+    const cutoffTime = new Date(now.getTime() - STUCK_GAP_THRESHOLD_HOURS * 60 * 60 * 1000);
+    const cutoffTimeISO = cutoffTime.toISOString();
+
+    logger.info(
+      `Starting cleanup of stuck in-progress gaps updated before ${cutoffTimeISO} (max ${MAX_GAPS_TO_PROCESS} gaps)`
+    );
+
+    // Get rule IDs with gaps that have in_progress_intervals (limit to MAX_RULES_TO_PROCESS)
+    const { ruleIds: allRuleIds } = await getRuleIdsWithGaps(rulesClientContext, {
+      start: startDate.toISOString(),
+      end: now.toISOString(),
+      hasInProgressIntervals: true,
+    });
+
+    if (allRuleIds.length === 0) {
+      logger.debug('No rules with in-progress gaps found');
+      return;
+    }
+
+    const ruleIdsToProcess = allRuleIds.slice(0, MAX_RULES_TO_PROCESS);
+    logger.info(
+      `Processing ${ruleIdsToProcess.length} rules (out of ${allRuleIds.length} total) for stuck gap cleanup`
+    );
+
+    const gapsResponse = await findGapsSearchAfter({
+      eventLogClient,
+      logger,
+      params: {
+        ruleIds: ruleIdsToProcess,
+        start: startDate.toISOString(),
+        end: now.toISOString(),
+        perPage: MAX_GAPS_TO_PROCESS,
+        hasInProgressIntervals: true,
+        updatedBefore: cutoffTimeISO,
+        sortField: '@timestamp',
+        sortOrder: 'asc',
+      },
+    });
+
+    const gapsToCheck = gapsResponse.data;
+
+    // Close PIT if it was created
+    if (gapsResponse.pitId) {
+      await eventLogClient.closePointInTime(gapsResponse.pitId);
+    }
+
+    if (gapsToCheck.length === 0) {
+      logger.debug('No stuck in-progress gaps found');
+      return;
+    }
+
+    logger.info(`Found ${gapsToCheck.length} gaps to check for stuck in-progress intervals`);
+
+    const gapsWithoutBackfills = await filterGapsWithOverlappingBackfills(
+      gapsToCheck,
+      rulesClientContext,
+      (message) => logger.debug(message)
+    );
+
+    // Build a set of gaps that require reset
+    const withoutBackfillsSet = new Set(
+      gapsWithoutBackfills
+        .filter((gap) => gap.internalFields !== undefined)
+        .map((gap) => gap.internalFields?._id)
```

Comment:
first it filters internalFields which are undefined, but then allows undefined value in subsequent map function.

filter seems redundant here.

```suggestion
        .map((gap) => gap?.internalFields?._id)
```

or something like this should be in place

```suggestion
        .filter((gap) => gap?.internalFields?._id)
        .map((gap) => gap.internalFields?._id)
```

Or reduce